COMPONENT-API.md

```diff
@@ -1,376 +1,864 @@
 
-```javascript
-module.exports = {
-  name: "name", // required
-  version: "0.0.1", // required
-  props,
-  methods,
-  run(event) {
-    console.log("Run any Node.js code here");
-  }
-};
+- Deploy components to Pipedream using the CLI
+- Invoke a component manually, or on a schedule or HTTP request
+- Maintain state across component invocations
+- Emit deduped events using the `unique` and `greatest` strategies
+- Use Pipedream managed OAuth for an app
+- Use npm packages in components
+
+## Getting Started with the CLI
+
+Several examples below use the CLI. To install it, run:
+
+```bash
+curl https://cli.pipedream.com/install | sh
 ```
 
-The component API borrows concepts and structure from frontend components, like those in [Vue.js](https://vuejs.org/) or [React](https://reactjs.org/).
+Pipedream publishes a build of the CLI for macOS and Linux (for `386`, `amd64`, `arm`, and `arm64` architectures). If you need to use the CLI on another OS or architecture, [please reach out](https://docs.pipedream.com/support/).
 
-Components [run on Pipedream infrastructure](#how-components-run).
+Once the CLI is installed, log into Pipedream:
 
-## Features
+```bash
+pd login
+```
 
-Component can:
+See the [CLI reference](https://docs.pipedream.com/cli/reference/) for detailed usage and examples beyond those covered below.
 
-- Accept input via [props](#props)
-- Define [methods](#methods).
-- [`require` any npm package](#using-npm-packages)
-- Spin up Pipedream infrastructure, like HTTP servers and scheduled jobs, by declaring the correct [interfaces](#interfaces)
-- Store and retrieve state using the [built-in key-value store](#servicedb)
-- [Emit data](#thisemit) you process within the component, allowing you to access it outside of Pipedream via API.
+## Contributing
 
-## Reference
+Deploy or contribute to curated open source components in Pipedream's Github repo. Or author, deploy and maintain your own via your standard CI/CD process.
 
-<!--ts-->
+# Component Lifecycle
 
-- [name](#name)
-  - [Name slugs](#name-slugs)
-- [version](#version)
-- [run method](#run-method)
-- [Referencing this](#referencing-this)
-  - [Referencing props](#referencing-props)
-  - [Referencing methods](#referencing-methods)
-- [Interfaces](#interfaces)
-  - [`$.interface.http`](#interfacehttp)
-    - [Methods on the HTTP interface](#methods-on-the-http-interface)
-  - [`$.interface.timer`](#interfacetimer)
-- [props](#props)
-  - [Example](#example)
-- [methods](#methods)
-- [`this.$emit`](#thisemit)
-  - [How to emit events](#how-to-emit-events)
-  - [Retrieving events programmatically](#retrieving-events-programmatically)
-- [`$.service.db`](#servicedb)
-  - [set method](#set-method)
-  - [get method](#get-method)
-- [Using npm packages](#using-npm-packages)
-- [How components run](#how-components-run)
+## Diagram
 
-<!--te-->
+![image-20200816223906765](./images/image-20200816223906765.png)
```

Comment:
When I view this file in the Github UI, this image didn't render - is that expected?

Reply:
no -- it worked in my personal repo before i merged. i will fix. thanks!